crates/nu-protocol/src/engine/ctrlc.rs

```diff
@@ -0,0 +1,119 @@
+use std::fmt::Debug;
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc, Mutex,
+};
+
+pub type Handler = Box<dyn Fn() + Send + Sync>;
+
+#[derive(Clone)]
+pub struct Handlers {
+    handlers: Arc<Mutex<Vec<(u64, Handler)>>>,
+    // we use an u64 so an overflow is impractical
+    next_id: Arc<AtomicU64>,
```

Comment:
Since this is in plugin code anyway, you might as well just use `nu_plugin_core::util::Sequence`

That uses `AtomicUsize` since `AtomicU64` isn't always available (although usually is) but will still be 64-bit on 64-bit platforms. Alternatively I guess we could use something like `atomic_shim` to cover the cases where that isn't available

Reply:
ah: nu-plugin-core has nu-protocol in its Cargo.toml, which makes it impractical to use nu_plugin_core::util::Sequence in nu-protocol. I could move nu_protocol::engine::ctrlc to nu_plugin_core? If so, do you think under util would be a good fit?

Reply:
It needs to be part of `EngineState`, right? So it has to be in at least `nu-protocol` or before.

Reply:
@devyn I've pushed moving sequence to nu-protocol::engine: What do you think? I don't love it: but it's nice that Sequence.next can return a `ShellError`